javaslang/src/main/java/javaslang/collection/Map.java

```diff
@@ -387,6 +408,26 @@ default int length() {
     <V2> Map<K, V2> mapValues(Function<? super V, ? extends V2> valueMapper);
 
     /**
+     * Calculates the <em>maximum</em> of this {@code Map} values using the natural {@code Comparator} of the values, if exists.
+     *
+     * @return {@code Some(maximum)} of this elements or {@code None} if this is empty or no {@code Comparator} is applicable
+     */
+    @SuppressWarnings("unchecked")
+    @Override
+    default Option<Tuple2<K, V>> max() {
+        if (isEmpty()) {
+            return Option.none();
+        } else {
+            if (head()._2 instanceof Comparator) {
```

Comment:
Multimap does need similar overrides, I will add them, too.
Also the coverage decreased. I will take care of it.